contracts/schemes/JoinAndQuit.sol

```diff
@@ -0,0 +1,243 @@
+pragma solidity ^0.5.16;
+
+import "../votingMachines/VotingMachineCallbacks.sol";
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+import "../libs/StringUtil.sol";
+
+
+/**
+ * @title A scheme for join in a dao.
+ * - A member can be proposed to join in by sending a min amount of fee.
+ * - A member can ask to quite (RageQuit) a dao on any time.
+ * - A member can donate to a dao.
+ */
+contract JoinAndQuit is
+        VotingMachineCallbacks,
+        ProposalExecuteInterface,
+        Initializable {
+    using SafeMath for uint;
+    using SafeERC20 for address;
+    using StringUtil for string;
+
+    event JoinInProposal(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        string _descriptionHash,
+        address _proposedMember,
+        uint256 _fundAmount
```

Comment:
We are calling it "fee" when we write "minFeeToJoin". Let's be consistent about this. 

I think "funding" is a cbit confusing here..  here it is about  money flowing into the DAO, but we are also using the term for money flowing OUT of the DAO (as in "funding request"). 

We can use "fee" - so instead of `_fundAmont` just write `_fee` or `_feeAmount`.

Reply:
Or "donataion" ,as in

`minDonationToJoin`
`donations` --> the mapping that is now called `fundings`
`_donationAmount`